spec/system/publishers/publishers_can_manage_self_disclosure_spec.rb

```diff
@@ -159,26 +160,41 @@
        .and not_change(SelfDisclosure, :count)
     end
 
-    describe "visit jobseeker self disclosure form" do
-      let(:request) { create(:self_disclosure_request, job_application_id: job_application.id) }
-
-      before { request.manual! }
+    describe "visit jobseeker self disclosure form", :versioning do
+      before do
+        create(:self_disclosure_request, :manual, job_application_id: job_application.id)
+        publisher_ats_self_disclosure_page.load(
+          vacancy_id: vacancy.id,
+          job_application_id: job_application.id,
+        )
+      end
 
       context "when request is pending" do
+        it "handles errors" do
+          publisher_ats_self_disclosure_page.goto_references_and_self_disclosure_form.click
+          click_on "Save and continue"
+          expect(publisher_ats_self_disclosure_page.errors.map(&:text))
+            .to eq(["Select yes if you would like to collect self disclosure through the service"])
+        end
+
+        it "copes with saying no" do
```

Comment:
I think something like "takes jobseeker back to pre interviews check if they choose not to collect self disclosure through the service" would make this test a bit more clear. It feels like there is something else we can assert other than the page the user lands on in this spec.